app/grandchallenge/reader_studies/views.py

```diff
@@ -1430,50 +1432,16 @@ def form_valid(self, form):
         """
         instance = self.get_object()
         assigned_civs = []
-        for ci_slug, civ in form.cleaned_data.items():
-            if ci_slug == "order":
+        for ci_slug, new_value in form.cleaned_data.items():
+            if ci_slug == "order" or new_value is None:
                 continue
-            ci = ComponentInterface.objects.get(slug=ci_slug)
-            current_value = instance.values.filter(interface=ci).first()
-
-            if ci.is_json_kind and not ci.requires_file:
-                if current_value:
-                    assigned_civs.append(current_value)
-                val = ComponentInterfaceValue.objects.create(
-                    interface=ci, value=civ
-                )
-                instance.values.add(val)
-            else:
-                if civ is None:
-                    # Value is cleared
-                    if current_value:
-                        assigned_civs.append(current_value)
-                    continue
-
-                # If there is already a value for the provided civ's interface in
-                # this display set, remove it from this display set. Cast to list
-                # to evaluate immediately.
-                assigned_civs += list(
-                    instance.values.exclude(pk=civ.pk).filter(interface=ci)
-                )
-
-                # Add the provided civ to the current display set
-                instance.values.add(civ)
-
-        # Create a new display set for any image civs that have been replaced by a
-        # new value in this display set, to ensure it remains connected to
-        # the reader study.
-        for assigned in assigned_civs:
-            if assigned.interface.is_image_kind and (
-                not instance.reader_study.display_sets.exclude(pk=instance.pk)
-                .filter(values=assigned)
-                .exists()
-            ):
-                ds = DisplaySet.objects.create(
-                    reader_study=instance.reader_study
-                )
-                ds.values.add(assigned)
```

Comment:
I'm not sure if this is necessary or even desired behaviour. We don't do this for archive items either. The only reason I can see is that a user loses view rights for an image when its no longer connected to the ds (and also not to an archive or job etc). But then for consistency, this should also happen for archive items, shouldn't it? Or maybe we could consider not removing the view permission for the original uploader of the image (image origin --> imageuploadsession.creator always keep the view permission)?

Reply:
I think that it's fine to remove.